File: 백준/Bronze/9506. 약수들의 합/약수들의 합.java
import java.util.ArrayList;
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        while (true) {
            ArrayList<Integer> factorList = new ArrayList<>();

            int num = sc.nextInt();

            if (num == -1) {
                break;
            }

            int sum = 0;
            for (int i = 2; i < num; i++) {
                if (num % i == 0) {
                    factorList.add(i);
                    sum += i;
                }
            }

            if (num == sum + 1) {
                System.out.print(num + " = 1");
                for ( int factor : factorList) {
                    System.out.print(" + " + factor);
                }

                System.out.println();
            } else {
                System.out.println(num + " is NOT perfect.");
            }
        }
    }
}

File: 백준/Bronze/9506. 약수들의 합/약수들의 합.cc
#include <iostream>
#include <list>
using namespace std;

int main() {

	while (true)
	{
		list<int> factorList;
		int num;
		cin >> num;

		if (num == -1)
		{
			return 0;
		}

		int sum = 1;
		for (int i = 2; i < num; i++)
		{
			if (num % i == 0)
			{
				factorList.push_back(i);
				sum += i;
			}
		}

		if (num == sum)
		{
			cout << num << " = 1";
			for (int factor : factorList)
			{
				cout << " + " << factor;
			}

			cout << endl;

		} else {
			cout << num << " is NOT perfect." << endl;
		}
	}
}